From 9c038a6638f0a97f6d8a819df3a5a372828f7c52 Mon Sep 17 00:00:00 2001
From: "magomez@igalia.com"
 <magomez@igalia.com@268f45cc-cd09-0410-ab3c-d52691b4dbfc>
Date: Wed, 16 Aug 2017 11:26:39 +0000
Subject: [PATCH] [GTK] Crash of WebProcess with on-demand AC
 https://bugs.webkit.org/show_bug.cgi?id=171161

Reviewed by Carlos Garcia Campos.

Set the viewOverlayRootLayer to the previous layerTreeHost if it exists. This avoids a
crash when reusing the previous layerTreeHost, because it would keep a reference to an
already destroyed viewOverlayRootLayer.

* WebProcess/WebPage/AcceleratedDrawingArea.cpp:
(WebKit::AcceleratedDrawingArea::attachViewOverlayGraphicsLayer):


git-svn-id: http://svn.webkit.org/repository/webkit/trunk@220793 268f45cc-cd09-0410-ab3c-d52691b4dbfc

diff --git a/Source/WebKit/WebProcess/WebPage/AcceleratedDrawingArea.cpp b/Source/WebKit/WebProcess/WebPage/AcceleratedDrawingArea.cpp
index 8e831ceb878..96c37cf5ee2 100644
--- a/Source/WebKit2/WebProcess/WebPage/AcceleratedDrawingArea.cpp
+++ b/Source/WebKit2/WebProcess/WebPage/AcceleratedDrawingArea.cpp
@@ -474,6 +474,8 @@ void AcceleratedDrawingArea::attachViewOverlayGraphicsLayer(Frame* frame, Graphi
 
     if (m_layerTreeHost)
         m_layerTreeHost->setViewOverlayRootLayer(viewOverlayRootLayer);
+    else if (m_previousLayerTreeHost)
+        m_previousLayerTreeHost->setViewOverlayRootLayer(viewOverlayRootLayer);
 }
 
 } // namespace WebKit
-- 
2.13.5

